### PR TITLE
Add kamon-pekko-http

### PR DESCRIPTION
**Pull request by DieBauer**
Since Akka changed its license to BSL, an Apache fork of Akka was made: [Apache Pekko](https://github.com/apache/incubator-pekko) (and [Apache Pekko Http](https://github.com/apache/incubator-pekko-http)).

An initial release is released, and the major change is that the namespaces changed from akka. to org.apache.pekko..

This PR adds the first module that copy the behavior of kamon-akka-http.

The changes can be summed up as follows:

    Change the package imports to org.apache.pekko.
    Change configuration keys from akka to pekko.
    Rewrite deprecated constructs

All tests are green.

This builds on top of #1264

### CODE REVIEW

**Comment by pjfanning** [comment]
@DieBauer Pekko HTTP 1.0.0 should be out within the week. Pekko Connectors Kafka (alpakka-kafka equivalent) should be out this week too - if you are interested in adding support for that too.

File: instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    .intercept(method("apply"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+  onType("org.apache.pekko.http.scaladsl.server.directives.RouteDirectives")
+    .intercept(method("complete"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("redirect"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("failWith"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+
+  /**
+    * Akka-http 10.1.x compatibility.
```
Comment (pjfanning) [inline instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala:82]: does pekko-http need this?

File: instrumentation/kamon-pekko-http/src/main/scala-2.12/kamon/instrumentation/pekko/http/PekkoHttpServerInstrumentation.scala [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+  onType("org.apache.pekko.http.scaladsl.server.directives.RouteDirectives")
+    .intercept(method("complete"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("redirect"), classOf[ResolveOperationNameOnRouteInterceptor])
+    .intercept(method("failWith"), classOf[ResolveOperationNameOnRouteInterceptor])
+
+  /**
+    * Akka-http 10.1.x compatibility.
+    */
+
+  onType("org.apache.pekko.http.scaladsl.Http2Ext")
```
Comment (pjfanning) [inline instrumentation/kamon-pekko-http/src/main/scala-2.12/kamon/instrumentation/pekko/http/PekkoHttpServerInstrumentation.scala:102]: again, pekko-http may not need this - there is just a v1.0.0 to handle (based on akka-http 10.2)

File: instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala [hunk 3]
```diff
@@ -0,0 +1,330 @@
+package kamon.instrumentation.pekko.http
```
Comment (pjfanning) [inline instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala:1]: can you fix the directory? - still in `akka` directory

Comment (DieBauer) [reply instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala]: Nice catch. done.

File: instrumentation/kamon-pekko-http/src/main/scala-2.12/kamon/instrumentation/pekko/http/PekkoHttpServerInstrumentation.scala (hunk 2 above)
Comment (DieBauer) [reply instrumentation/kamon-pekko-http/src/main/scala-2.12/kamon/instrumentation/pekko/http/PekkoHttpServerInstrumentation.scala]: Correct! that package is not there in Pekko HTTP. `org.apache.pekko.http.impl.engine.http2.Http2Ext` is, which is instrumented.

File: instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala (hunk 1 above)
Comment (DieBauer) [reply instrumentation/kamon-pekko-http/src/main/scala-2.13/kamon/instrumentation/akka/http/PekkoHttpServerInstrumentation.scala]: Correct! that package is not there in Pekko HTTP. `org.apache.pekko.http.impl.engine.http2.Http2Ext` is, which is instrumented.

**Comment by pjfanning** [comment]
@DieBauer Pekko HTTP 1.0.0 is released

**Comment by ivantopo** [comment]
@DieBauer could you please rebase this PR on master now that the pekko instrumentation is merged? thanks!

**Comment by DieBauer** [comment]
Done!

**Comment by ivantopo** [comment]
Thank you very much for all the work here @DieBauer! I'll release this as Kamon Telemetry 2.7 at some point before Sunday since I have one more bit of instrumentation I want to include in that release. Great job!

**Comment by drenizg** [comment]
Hi everyone, nice job! Is there any update on when is this being released?

**Comment by mebigfatguy** [comment]
Sorry for  being such a scala noob, but  trying to build the 2.13 version of kamon-pekko, kamon-pekko-http

and get

[info] loading project definition from /home/acme/dev/external/Kamon/instrumentation/kamon-pekko/project
/home/acme/dev/external/Kamon/instrumentation/kamon-pekko/build.sbt:4: error: not found: value scala_2.12_version
  crossScalaVersions := Seq(`scala_2.12_version`, `scala_2.13_version`)
                            ^
/home/acme/dev/external/Kamon/instrumentation/kamon-pekko/build.sbt:4: error: not found: value scala_2.13_version
  crossScalaVersions := Seq(`scala_2.12_version`, `scala_2.13_version`)


could  someone point me in the right direction?

**Pull request merged**